build/templates/binary-phar-autoload.php.in

```diff
@@ -30,21 +30,36 @@ if (version_compare('7.3.0', PHP_VERSION, '>')) {
     die(1);
 }
 
-foreach (['dom', 'json', 'libxml', 'mbstring', 'tokenizer', 'xml', 'xmlwriter'] as $extension) {
-    if (extension_loaded($extension)) {
-        continue;
-    }
+(static function (): void {
```

Comment:
Not sure if you like that, but I am using a self-invoking function here to contain the variables. Alternatively I could reset them after using them.

Reply:
Also, what about the duplication, perhaps we should extract and reuse this code?